02_erc20/contract.etch

```diff
@@ -84,27 +143,57 @@ function balanceOf(owner: Address) : UInt256
     return UInt256(0u64);
   endif
 
-  return balance_state.get();
+  return balance_state.get(UInt256(0u64));
 endfunction
 
 // transfer(address _to, uint256 _value) public returns (bool success) [Send _value amount of tokens to address _to]
+// Note that the Fetch ledger does not have an implicit sender address. Hence, we need a different function signature
+@action
+function transfer(sender: Address, to: Address, value: UInt256) : Bool
```

Comment:
To make transition from Ethereum world easier I suggest using `sender` as the new variable (in reference to `msg.sender` which is available by default there)